ci/hardware_tests/device/legacy.py

```
@@ -15,20 +16,23 @@ def touch(self, location, action):
 
     def update_firmware(self, file=None):
         if file:
+            run(f"headertool {file}", shell=True, check=True)  # check if file exists
```

Ruff (0.15.18)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)